### PR TITLE
Moved setting net service into 1st stage

### PR DESCRIPTION
**Pull request by mchf**
bnc#955260

### CODE REVIEW

File: package/yast2-network.changes [hunk 1]
```diff
@@ -1,4 +1,10 @@
 -------------------------------------------------------------------
+Mon Feb 22 11:52:09 UTC 2016 - mfilka@suse.com
+
+- bnc#955260
+  - Moved setting network service into 1st stage
+- 3.1.140.2
```
Comment (mvidner) [inline package/yast2-network.changes]: The SP1 and master versioning has diverged, but you are applying a SP1 version to master.

File: src/lib/network/network_autoyast.rb [hunk 2]
```diff
@@ -35,6 +47,79 @@ def merge_configs(conf)
...
+
+      return if udev_rules.nil? || udev_rules.empty?
+
+      LanItems.Read
+
+      udev_rules.each do |rule|
+        name_to = rule["name"]
+        attr = rule["rule"]
+        key = rule["value"].downcase
+        item, matching_item = LanItems.Items.find { |_, i| i["hwinfo"]["busid"].downcase == key || i["hwinfo"]["mac"].downcase == key }
```
Comment (mvidner) [inline src/lib/network/network_autoyast.rb]: This line is too long

File: test/network_autoyast_test.rb [hunk 3]
```diff
@@ -148,4 +148,66 @@
...
+    Yast.import "NetworkService"
+    Yast.import "Mode"
+
+    let(:network_autoyast) { Yast::NetworkAutoYast.instance }
+
+    before(:each) do
+      allow(Yast::Mode).to receive(:autoinst).and_return(true)
+    end
+
+    def product_use_nm(nm_used)
```
Comment (mvidner) [inline test/network_autoyast_test.rb]: I think the spec is more readable if helpers like this are named with the stub/mock prefix, like `allow_product_use_nm`, `allow_networking_section`.

File: package/yast2-network.changes (hunk 1 above)
Comment (mchf) [inline package/yast2-network.changes]: ok f*ck ... bad target for merging. Thanks

File: src/lib/network/network_autoyast.rb (hunk 2 above)
Comment (mchf) [inline src/lib/network/network_autoyast.rb]: ok, I can do something with that

File: test/network_autoyast_test.rb (hunk 3 above)
Comment (mchf) [inline test/network_autoyast_test.rb]: ok

**Comment by mvidner** [comment]
Really [bsc#955260](https://bugzilla.suse.com/show_bug.cgi?id=955260)? "autoyast install over ssh with X-forwarding halts in second stage on restarting services" is marked as Verified since 2 months.

**Comment by mvidner** [comment]
I see, the card is https://trello.com/c/SUdZnkzy/416-5-sp2-finish-moving-device-renaming-into-first-stage-in-case-of-ay-installation .
Anyway, since a PR cannot change its target branch, we should continue in a new PR.

**Comment by mchf** [comment]
ok close for now and start again

**Pull request closed without merging**